src/toml_file.rs

```diff
@@ -27,3 +27,7 @@ pub fn read_toml_file<A: DeserializeOwned>(path: impl AsRef<Path>) -> Result<A,
     let contents = fs::read_to_string(path)?;
     Ok(toml::from_str(&contents)?)
 }
+
+pub fn from_string<A: DeserializeOwned>(contents: String) -> Result<A, TomlFileError> {
```

Comment:
`toml_file::from_string` makes no sense to me semantically. My best guess was that it creates a new file from a `String`, but where does TOML come into the equation? 

If we need this somewhere, we should just use `toml::from_str(&contents)` directly as it already does what we want.

Reply:
I did this for murky reasons before. I think I had a hard time getting the code to generate a `libcnb::Error` and that was one way.

I'm able to get this to compile and I think it does what I want:

```rs
                BuildpackTomlArg::Str(contents) => toml::from_str(&contents)
                    .map_err(TomlFileError::TomlDeserializationError)
                    .map_err(Error::CannotReadBuildpackDescriptor),
```

It's not perfectly true since the message in deserialization error says that it couldn't deserialize the file. It will need to be updated. I'm guessing if we decide to keep the feature we'll have to add an error enum to `error.rs`

Reply:
I updated to use `toml` directly as noted before.